crates/cli/src/main.rs

```diff
@@ -1,38 +1,57 @@
 mod bytecode;
+mod codegen;
 mod commands;
-mod exports;
 mod js;
-mod wasm_generator;
 mod wit;
 
+use crate::codegen::{DynamicGenerator, StaticGenerator, WitOptions};
 use crate::commands::{Cli, Command, EmitProviderCommandOpts};
-use crate::wasm_generator::r#static as static_generator;
-use anyhow::{bail, Result};
+use anyhow::Result;
 use clap::Parser;
+use codegen::CodeGenBuilder;
 use js::JS;
 use std::fs;
 use std::fs::File;
 use std::io::Write;
-use wasm_generator::dynamic as dynamic_generator;
 
 fn main() -> Result<()> {
     let args = Cli::parse();
 
-    match args.command {
-        Command::EmitProvider(opts) => emit_provider(&opts),
-        Command::Compile(opts) => {
+    match &args.command {
+        Command::EmitProvider(opts) => emit_provider(opts),
+        c @ Command::Compile(opts) | c @ Command::Build(opts) => {
+            if c.is_compile() {
+                println!(
```

Comment:
I think this should be emitted to stderr using `eprintln!` instead of stdout similar to how `--help` works.